libraries/app/api.cpp

```diff
@@ -303,25 +303,29 @@ namespace graphene { namespace app {
        FC_ASSERT( limit <= 100 );
        vector<operation_history_object> result;
        const auto& stats = account(db).statistics(db);
-       if( stats.most_recent_op == account_transaction_history_id_type() ) return result;
        const account_transaction_history_object* node = &stats.most_recent_op(db);
        if( start == operation_history_id_type() )
           start = node->operation_id;
+       if( start.instance.value > node->operation_id.instance.value )
+          start = node->operation_id;
 
-       while(node && node->operation_id.instance.value > stop.instance.value && result.size() < limit)
+       const auto& hist_idx = db.get_index_type<account_transaction_history_index>();
+       const auto& by_op_idx = hist_idx.indices().get<by_op>();
+       auto index_end = by_op_idx.end();
+       auto index_start = by_op_idx.begin();
+       auto itr = by_op_idx.lower_bound(boost::make_tuple(account, start));
+
+       while(itr != index_end && itr != index_start && itr->operation_id.instance.value > stop.instance.value && result.size() < limit)
```

Comment:
What if `0` has been removed from the database (e.g. due to `max-ops-per-account`)? In this case, `index_start` is skipped here, but it didn't get processed after the loop.

`itr != index_end` is redundant here, since line 306~310 will imply it: you know there is at least a node, you know start is not beyond the last node, so itr won't be at the end.